control-base/src/test/java/fi/nls/oskari/control/layer/GetWFSLayerFieldsHandlerTest.java

```diff
@@ -0,0 +1,92 @@
+package fi.nls.oskari.control.layer;
+
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.map.layer.OskariLayerServiceMybatisImpl;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceException;
+import fi.nls.oskari.util.WFSGetLayerFields;
+import fi.nls.test.control.JSONActionRouteTest;
+import fi.nls.test.util.ResourceHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({WFSGetLayerFields.class, OskariComponentManager.class})
+public class GetWFSLayerFieldsHandlerTest extends JSONActionRouteTest {
+    private static final Integer WFS_LAYER_ID = 1;
+    private static final String PARAM_LAYER_ID = "layer_id";
+    private static final GetWFSLayerFieldsHandler handler = new GetWFSLayerFieldsHandler();
+
+    @BeforeClass
+    public static void setup() throws JSONException, ServiceException {
+        mockWFSGetLayerFields();
+        mockOskariComponentManager();
+        handler.init();
+    }
+
+    private static void mockWFSGetLayerFields() throws JSONException, ServiceException {
+        PowerMockito.mockStatic(WFSGetLayerFields.class);
+        final JSONObject mockFields = getMockFields();
+        when(WFSGetLayerFields.getLayerFields(any())).thenReturn(mockFields);
+    }
+
+    private static JSONObject getMockFields() throws JSONException {
+        final String rawFieldsStr = "{\"attributes\": {\"attr-name\": \"string\"}, \"geometryField\": \"geometry\"}";
+        return new JSONObject(rawFieldsStr);
+    }
+
+    private static void mockOskariComponentManager() throws JSONException {
+        OskariLayerService mockLayerService = getMockLayerService();
+        PowerMockito.mockStatic(OskariComponentManager.class);
+        when(OskariComponentManager.getComponentOfType(OskariLayerService.class)).thenReturn(mockLayerService);
+    }
+
+    private static OskariLayerService getMockLayerService() throws JSONException {
+        OskariLayerService mockLayerService = mock(OskariLayerServiceMybatisImpl.class);
+        OskariLayer wfsLayer = new OskariLayer();
+        wfsLayer.setType("WFS");
+        final String rawJsonStr = "{\"data\": {\"locale\": {\"fi\": {\"field-en\": \"field-fi\"}}}}";
```

Comment:
"field-en" might be a bit misleading here. Rename to something like `"test-attribute": "label for fi"` or similar.